src/modules/expression-evaluator/src/evaluateExpression.ts

```diff
@@ -243,8 +261,14 @@ const simplifyObject = (item: number | string | boolean | BasicObject) => {
 }
 
 // Abstraction for GraphQL database query using Fetch
-const graphQLquery = async (query: string, variables: object, connection: IGraphQLConnection) => {
-  const queryResult = await connection.fetch(connection.endpoint, {
+const graphQLquery = async (
+  url: string,
+  query: string,
+  variables: object,
+  connection: IGraphQLConnection
+) => {
+  const endpoint = url !== 'graphQLEndpoint' && url !== '' ? url : connection.endpoint
```

Comment:
Not a biggie, but maybe make `graphQLEndpoint` a case-insensitive requirement. So `graphqlendPOINT` should also be acceptable. But not super important if you can't be bothered for now.

Nice to have the empty string in there too. :) I wonder if we should also accept `null` for a default? (No need to worry about `undefined` as you can't pass `undefined` in a JSON string)

Reply:
Sure, I'll add these 2 options, but not specific tests, since this will be just a prevention for missusage

Reply:
Done: 152e1e5

Checked that the case insensitive works fine using the "Evaluate expression gui", but couldn't test using `null`:
<img width="960" alt="Screen Shot 2021-04-28 at 10 07 49 AM" src="https://user-images.githubusercontent.com/16461988/116319051-ed0a9d80-a809-11eb-8259-8a6db6dc224c.png">

Reply:
Oh -- I can see the problem: 
`const endpoint = url.toLowerCase() !== 'graphqlendpoint' && url !== '' && url !== null`

When `url` is `null`, that first evaluation `url.toLowerCase()` is throwing an error. It works as expected if you put the null check first:
`const endpoint = url !== null && url.toLowerCase() !== 'graphqlendpoint' && url !== ''`

![Screen Shot 2021-04-28 at 11 05 42 AM](https://user-images.githubusercontent.com/5456533/116323238-b042a480-a811-11eb-963b-d83e0f2d9886.png)

:)

Reply:
Thanks I'l update that